crypto/x509/x509_vfy.c

```diff
@@ -237,11 +237,29 @@ static int verify_chain(X509_STORE_CTX *ctx)
     return ok;
 }
 
+int X509_STORE_CTX_verify(X509_STORE_CTX *ctx)
+{
+    int ret;
+
+    if (ctx == NULL) {
+        ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
+        return -1;
+    }
+    if (ctx->cert == NULL && sk_X509_num(ctx->untrusted) >= 1)
+        ctx->cert = sk_X509_value(ctx->untrusted, 0);
+    ret = X509_verify_cert(ctx);
+    ctx->cert = NULL;
```

Comment:
I still do not see the need to do this because although X509_STORE_CTX_cleanup won't clean this up, you have to do X509_STORE_CTX_init before you can reuse the X509_STORE_CTX and that one will overwrite the ctx->cert anyway.

Reply:
Ah, right!
Then I wonder about the sense of `X509_STORE_CTX_cleanup()` 
since `X509_STORE_CTX_init()` is then still needed. 

Anyway, I've removed the code that restores `ctx->cert`, which simplifies things.

Reply:
The X509_STORE_CTX_cleanup() was needed when X509_STORE_CTX was a public structure so you could just declare it on stack.

Reply:
Hmm, but `X509_STORE_CTX_init()` can be used regardless how the structure is allocated, 
and why not make  `X509_STORE_CTX_init()` call  `X509_STORE_CTX_cleanup()` automatically?
Then `X509_STORE_CTX_cleanup()` would not need to be called explicitly before that.
(It should not hurt if it is called in addition before that because it is idempotent.)

Reply:
I've implemented this idea in #14088.